### ISSUE TITLE
Unexpected Internal Software Issue (EX_SOFTWARE)

### ISSUE THREAD

**Issue by pubnoconst** [dotnet/templating#6347]
I was trying to create a uno project  but met with the following error.
```
$ dotnet new unoapp -o MyApp -mobile=false --skia-wpf=false --skia-linux-fb=false --vscode
Unable to cast object of type 'System.Boolean' to type 'System.String'.
   at System.CommandLine.Binding.ArgumentConverter.GetValueOrDefault[T](ArgumentConversionResult result)
   at Microsoft.TemplateEngine.Cli.Commands.TemplateOptionResult.FromParseResult(TemplateOption option, ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.TemplateResult.FromParseResult(TemplateCommand templateCommand, ParseResult parseResult)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.CollectTemplateMatchInfo(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.HandleNoTemplateFoundResult(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup, IReporter reporter)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.HandleTemplateInstantationAsync(InstantiateCommandArgs args, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, TemplateGroup templateGroup, CancellationToken cancellationToken)
   at Microsoft.TemplateEngine.Cli.Commands.InstantiateCommand.ExecuteIntAsync(InstantiateCommandArgs instantiateArgs, IEngineEnvironmentSettings environmentSettings, TemplatePackageManager templatePackageManager, InvocationContext context)
   at Microsoft.TemplateEngine.Cli.Commands.BaseCommand`1.InvokeAsync(InvocationContext context)
```

My uno-check

```
🚑 Uno Platform Check v1.12.1.0 💉
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
This tool will check your Uno Platform development environment.
If problems are detected, it will offer the option to try and fix them for you, or suggest a way to fix them yourself.
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
⏳ Synchronizing configuration... ok
⏳ Scheduling appointments... ok

🔎 .NET SDK Checkup...
  – 7.0.104 - /usr/lib64/dotnet/sdk/7.0.104

🔎 .NET SDK - Workload Deduplication Checkup...

🔎 GTK3+ Checkup...

🔎 Linux Ninja Build Checkup...
  ✔ Ninja Build Version: 1.10.2

🔎 .NET SDK - Workloads (7.0.102) Checkup...
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

✔ Congratulations, everything looks great!
```

I am running Fedora Linux 37.

**Comment by YuliiaKovalova** [comment]
@GangWang01 , could you take a look?

**Comment by GangWang01** [comment]
From uno-check the issue happened to dotnet 7.0.104.
I checked the template with short name `unoapp` in [Uno.Templates](https://www.nuget.org/packages/Uno.Templates) with multiple versions (latest one `4.9.0-dev.107`, 4.9.0-dev.53, 4.8.8, 4.8.0-dev.309, oldest one `4.8.0-dev.260`). I didn't find symbols for the following options specified in template.json or override alias in dotnetcli.host.json.

- mobile
- skia-wpf
- skia-linux-fb

Based on the info above, it looks like the same issue as #5725. With dotnet 7.0.1xx, if invalid options are given and there is boolean option this issue will happen. This issue was fixed for 7.0.2xx+.

@pubnoconst could you please confirm the following things? Then we can understand if it's the same issue and know what to do next step for further investigation.

1. Check if the options given in the command are valid by running `dotnet new unoapp -h`.
2. If given options are valid, please provide the version of installed package uno.templates Uno.Templates. You can get the installed version by running `dotnet new uninstall`.

Thank you!

**Comment by YuliiaKovalova** [comment]
Close the ticket since no response from the customer.

Feel free to reopen after providing the requested details.
